vcd/resource_vcd_org_vdc.go

```diff
@@ -0,0 +1,502 @@
+					Description: "Capacity used. If the VDC AllocationModel is ReservationPool, this number represents the percentage of the reservation that is in use. For all other allocation models, it represents the percentage of the allocation that is in use.",
+				},
+				"overhead": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Number of Units allocated to system resources such as vShield Manager virtual machines and shadow virtual machines provisioned from this Provider VDC.",
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceVcdVdcCreate,
+		Delete: resourceVcdVdcDelete,
+		Read:   resourceVcdVdcRead,
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization to create the VDC in",
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"allocation_model": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AllocationVApp", "AllocationPool", "ReservationPool"}, false),
+				Description:  "The allocation model used by this VDC; must be one of {AllocationVApp, AllocationPool, ReservationPool}",
+			},
+			"compute_capacity": &schema.Schema{
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cpu":    &capacityWithUsage,
+						"memory": &capacityWithUsage,
+					},
+				},
+				Description: "The compute capacity allocated to this VDC.",
+			},
+			"nic_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Maximum number of virtual NICs allowed in this VDC. Defaults to 0, which specifies an unlimited number.",
+			},
+			"network_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Maximum number of network objects that can be deployed in this VDC. Defaults to 0, which means no networks can be deployed.",
+			},
+			"vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The maximum number of VMs that can be created in this VDC. Includes deployed and undeployed VMs in vApps and vApp templates. Defaults to 0, which specifies an unlimited number.",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     true,
+				Description: "True if this VDC is enabled for use by the organization VDCs. Default is true.",
+			},
+			"storage_profile": &schema.Schema{
+				Type:     schema.TypeList,
```

Comment:
Worth adding for better validation and avoiding index out of range panics.
```
MaxItems     1,
MinItems      1
```

Reply:
Added `MinItems` only. And yet we have inside check too for that

Reply:
Great. I initially forgot multiple storage profiles possible initially, but that makes sense. Thanks!